src/clients/host-client.js

```diff
@@ -166,13 +156,15 @@ class HostClient extends BaseEvernodeClient {
         // Check whether is there any missed NFT sell offer that needs to be accepted
         // from the client-side in order to complete the registration.
         const registryAcc = new XrplAccount(this.config.registryAddress, null, { xrplApi: this.xrplApi });
-        const regNft = await this.getRegistrationNft();
-        if (!regNft) {
+        const regUriTokens = await this.getRegistrationUriTokens();
+        
+        if (!regUriTokens) {
             const regInfo = await this.getHostInfo(this.xrplAcc.address);
             if (regInfo) {
-                const sellOffer = (await registryAcc.getNftOffers()).find(o => o.NFTokenID == regInfo.nfTokenId);
+                const sellOffer = (await registryAcc.getURITokens()).find(o => o.index == regInfo.nfTokenId);
+                console.log('sell offer')
                 if (sellOffer) {
-                    await this.xrplAcc.buyNft(sellOffer.index);
+                    await this.xrplAcc.buyURIToken(sellOffer.index, XrplConstants.MIN_XRP_AMOUNT, XrplConstants.XRP);
```

Comment:
What if we change `buyURIToken(uriTokenID, amount, currency, issuer = null, options = {})` to `buyURIToken(uriToken, issuer = null, options = {})`?
I think then we can directly use `uriToken.Amount` for Amount in `URITokenBuy` transaction